buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -95,42 +101,55 @@ class ClusterFormationTasks {
         }
 
         File pidFile = pidFile(baseDir)
-        Task start = project.tasks.create(name: "${task.name}#start", type: Exec, dependsOn: setup) {
-            workingDir home
-            executable 'sh'
-            List esArgs = [
-                'bin/elasticsearch',
-                '-d', // daemonize!
-                "-Des.http.port=${config.httpPort}",
-                "-Des.transport.tcp.port=${config.transportPort}",
-                "-Des.pidfile=${pidFile}",
-                "-Des.path.repo=${home}/repo",
-                "-Des.path.shared_data=${home}/../",
-            ]
-            esArgs.addAll(config.systemProperties.collect {key, value -> "-D${key}=${value}"})
-            args esArgs
-            errorOutput = new ByteArrayOutputStream()
-            doLast {
-                if (errorOutput.toString().isEmpty() == false) {
-                    logger.error(errorOutput.toString())
-                    new File(home, 'logs' + File.separator + clusterName + '.log').eachLine {
-                        line -> logger.error(line)
+        List esArgs = [
+            "-Des.http.port=${config.httpPort}",
+            "-Des.transport.tcp.port=${config.transportPort}",
+            "-Des.pidfile=${pidFile}",
+            "-Des.path.repo=${home}/repo",
+            "-Des.path.shared_data=${home}/../",
+        ]
+        esArgs.addAll(config.systemProperties.collect {key, value -> "-D${key}=${value}"})
+        Closure esPostStartActions = { ant, logger ->
+            ant.waitfor(maxwait: '30', maxwaitunit: 'second', checkevery: '500', checkeveryunit: 'millisecond', timeoutproperty: "failed${task.name}#start") {
+                and {
+                    resourceexists {
+                        file file: pidFile.toString()
                     }
-                    throw new GradleException('Failed to start elasticsearch')
+                    http(url: "http://localhost:${config.httpPort}")
                 }
-                ant.waitfor(maxwait: '30', maxwaitunit: 'second', checkevery: '500', checkeveryunit: 'millisecond', timeoutproperty: "failed${task.name}#start") {
-                    and {
-                        resourceexists {
-                            file file: pidFile.toString()
-                        }
-                        http(url: "http://localhost:${config.httpPort}")
-                    }
+            }
+            if (ant.properties["failed${task.name}#start"]) {
```

Comment:
The condition was not properly working before with
`if (ant.properties.containsKey("failed${task.name}#start")) {` because `containsKey` is called with a `GString` which does not equal to the represented `String`.